x-pack/platform/plugins/shared/onechat/common/step_types/run_agent_step.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { z } from '@kbn/zod/v4';
+import type { CommonStepDefinition } from '@kbn/workflows-extensions/common';
+
+/**
+ * Step type ID for the onechat run agent step.
+ */
+export const RunAgentStepTypeId = 'onechat.runAgent';
+
+/**
+ * Input schema for the run agent step.
+ */
+export const InputSchema = z.object({
+  /**
+   * The ID of the agent to chat with. Defaults to the default Elastic AI agent.
+   */
+  agent_id: z.string().optional(),
+  /**
+   * output schema for the run agent step, if provided agent will return structured output
+   */
+  schema: z.string().optional(),
+  /**
+   * The user input message to send to the agent.
+   */
+  input: z.string(),
```

Comment:
NIT: Similar, we will have `step.agent.input.input`, maybe better to use `message`?